feat(link): 大厅早期支持

Pigeon0v0 opened the pull request:
PCL CE 大厅暂时采用 EasyTier 作为联机核心，使用 Natayark ID 进行身份验证。

大厅目前仍然处于早期开发状态，预期会在 2025.8 之前发布第一个预览版。

目前还存在一些小问题，但也不是不能用：

- 支持 SOCKS5 以避免使用管理员权限，目前加入方需要用管理员身份打开启动器才可以进行联机
- EasyTier 无自带访问限制，目前启动器通过添加 Windows 防火墙实现控制，但用户若自行关闭防火墙则失效

<details>
<summary>截图预览</summary>

![image](https://github.com/user-attachments/assets/ee98d1df-d7b6-47ae-b6fb-1e26d54a05cb)

</details>

Pigeon0v0 committed:
feat(link): ET 功能补全

Pigeon0v0 committed:
feat: 虚假的 MC 服务端

Pigeon0v0 committed:
feat(link): Natayark ID Auth
dev: clean up unused imports

Pigeon0v0 committed:
Merge branch 'dev' into link-port

XiaoQiAWA commented:
![image](https://github.com/user-attachments/assets/da4398e5-0f38-4139-81e0-429072ac054a)
这玩意甚至会被火绒报毒

Pigeon0v0 commented:
> 这玩意甚至会被火绒报毒

这是主线更新带上来的问题，没辙...

Pigeon0v0 committed:
Merge branch 'dev' into link-port

Pigeon0v0 committed:
feat(link): 支持 Natayark ID 登录，添加了相关用户协议
feat(ci): Action 添加 Naid 相关密钥
imp(comp): 每次切回窗口时才读取剪贴板
imp: 调整最小窗体大小
dev: trim code

Pigeon0v0 commented:
目前其实已经能用了，有空的可以测试看看 = =

需要手动补充中继服务器

Pigeon0v0 committed:
fix: 未进行验证

Pigeon0v0 committed:
fix: ui

Pigeon0v0 committed:
Merge branch 'dev' into link-port

Pigeon0v0 committed:
Merge branch 'dev' into link-port

Pigeon0v0 committed:
Merge branch 'link-port' into link-new

Pigeon0v0 committed:
Merge pull request #7 from PCL-Community/link-new

Pigeon0v0 committed:
merge

Pigeon0v0 committed:
merge

tangge233 committed:
Merge branch 'dev' into link-port

Pigeon0v0 committed:
Merge branch 'dev' into link-port

Pigeon0v0 committed:
feat: 支持传输 NAID 信息与 MC 档案信息
dev: clean up

Pigeon0v0 committed:
Merge branch 'link-port' of https://github.com/PCL-Community/PCL2-CE-Pigeon into link-port

Pigeon0v0 committed:
feat(link): 支持创建防火墙规则，需要管理员权限
fix: 联机完成界面的网络质量检测现在是准确的了
imp: 局域网广播支持显示对方 ID
fix: 无 ET 进程时尝试停止可能抛出意外的异常
fix: Naid 登录时未能捕捉 401 异常
dev: reduce warnings
dev: clean up

Pigeon0v0 commented:
等 Pysio 那边把 POST 搞好就可以合了大概

Pigeon0v0 committed:
del: 移除了多余的设置选项

Pigeon0v0 commented:
> 问一下怎么用，主界面没有联机
> 
> 

现在用不了，等正式发布或者关注用户群 / B 站抢内测名额

Pigeon0v0 committed:
feat(link): 添加大厅公告系统
feat(link): 支持不使用默认的中继服务器
imp: 大厅部分文本更加用户友好
chore: 文本描述优化
dev: 改进代码可读性

Pigeon0v0 commented:
还剩下个 Web Server 回调，@shimoranla 说他能做，所以就等这个了（

LinQingYuu committed:
feat:自动处理授权回调

LinQingYuu committed:
Update ModYggdrasil.vb

Pigeon0v0 committed:
Update Plain Craft Launcher 2/Pages/PageSetup/PageSetupLink.xaml.vb

Pigeon0v0 committed:
Update Plain Craft Launcher 2/Pages/PageSetup/PageSetupLink.xaml.vb

Pigeon0v0 committed:
Update Plain Craft Launcher 2/Modules/ModYggdrasil.vb

Pigeon0v0 committed:
Merge pull request #8 from PCL-Community/link-port-backend

Pigeon0v0 committed:
fix: vbproj

LinQingYuu committed:
remove location header

LinQingYuu committed:
Update ModYggdrasil.vb

ruattd committed:
feat(link): 授权回调前端 (#9)

ruattd committed:
fix(link): 添加缺失的 html 文件

Pigeon0v0 committed:
Merge branch 'dev' into link-port

LinQingYuu committed:
pref:移除没有必要的检查

LinQingYuu committed:
chore:调整日志文本

《石山》

Pigeon0v0 committed:
feat: 支持设置 MC 使用的 IP 协议栈
feat: 支持设置 EasyTier 中继策略
fix: MC 端口转发失败不重试
imp: 检测到主机实例已关闭自动解散大厅
imp: 文本改善

Pigeon0v0 committed:
Merge branch 'link-port' of https://github.com/PCL-Community/PCL2-CE-Pigeon into link-port

Pigeon0v0 committed:
Merge branch 'dev' into link-port

Pigeon0v0 committed:
Merge branch 'dev' into link-port

Pigeon0v0 committed:
feat: 检查系统防火墙
imp: 自动换行

Pigeon0v0 committed:
imp: 私有模式部署

Pigeon0v0 committed:
imp: 文本优化

Silverteal committed:
取消发送ETNetworkSecret

Pigeon0v0 committed:
Merge pull request #10 from PCL-Community/Silverteal-patch-1

LinQingYuu committed:
Merge branch 'dev' into link-port

Pigeon0v0 committed:
Merge branch 'dev' into link-port

Pigeon0v0 committed:
dev: 代码可读性与可维护性改进
imp: 大厅密钥现在上锁了
ui: 文本改进

Pigeon0v0 committed:
fix: 大厅名称设置不正确

Pigeon0v0 committed:
Merge branch 'dev' into link-port

Pigeon0v0 committed:
fix: Secret 拼接错误

ruattd committed:
refactor(link): 重构 NAID 登录回调服务端

ruattd committed:
fix(link): 无法正确获取回调 URL 参数 & 前端 API 调用顺序有误

Pigeon0v0 committed:
Merge branch 'dev' into link-port

Pigeon0v0 committed:
Merge branch 'link-port' into link-port-refactor-oauth-callback

Pigeon0v0 committed:
Merge pull request #11 from PCL-Community/link-port-refactor-oauth-callback

Pigeon0v0 committed:
feat(link): 使用 localhost 作为回调地址
imp(ui): 调整元素间距

Pigeon0v0 committed:
Merge branch 'dev' into link-port

Pigeon0v0 committed:
feat: 允许复制 IP
fix: 防火墙规则写反了

Pigeon0v0 committed:
Merge branch 'dev' into link-port

Pigeon0v0 committed:
fix: ci
feat: 支持撤销协议授权

Pigeon0v0 committed:
feat: 加入鸣谢列表
feat: 摇号现在可以云控概率了
feat: 加入个人信息处理政策入口
bump: et 2.3.2

Pigeon0v0 committed:
Merge branch 'dev' into link-port

Pigeon0v0 committed:
fix: 公告没内容时还展示公告
imp: 联机组件下载进度

wuliaodexiaoluo committed:
Merge branch 'dev' into link-port

ruattd commented:
等下 构建怎么炸了

LinQingYuu committed:
fix(link): 添加缺少的编译文件

Pigeon0v0 committed:
Merge branch 'dev' into link-port

Pigeon0v0 committed:
fix: 部分验证环节有误

Pigeon0v0 committed:
imp: NAID 令牌过期后不再尝试刷新令牌
chore: 移除硬编码大厅服务器地址

Pigeon0v0 committed:
Merge branch 'dev' into link-port

Pigeon0v0 committed:
fix: ci

Pigeon0v0 committed:
fix: ci

MoYuan-CN commented on Plain Craft Launcher 2/Resources/oauth-complete.html:
```diff
@@ -0,0 +1,131 @@
+    let favicon = document.getElementById("favicon");
+    fetch("/assets/background").then(async response => {
+        if (!response.ok) {
+            console.warn(`Fetching background image returns ${response.status}`);
+            return;
+        }
+        const blob = await response.blob();
+        root.style.setProperty("--bg-root-url", `url(${URL.createObjectURL(blob)})`);
+    });
+    fetch("/assets/icon").then(async response => {
+        if (!response.ok) {
+            console.warn(`Fetching favicon image returns ${response.status}`);
+            return;
+        }
+        const blob = await response.blob();
+        favicon.href = URL.createObjectURL(blob);
+    });
+    fetch("/status").then(async response => {
+        if (!response.ok) {
+            console.warn(`Fetching status returns ${response.status}`);
+            return;
+        }
+        const status = await response.json();
+        if (status.success) {
+            title.innerText = `欢迎回来, ${status.username}！`;
+            content.innerText = "已完成登录授权，现在可以安全关闭浏览器窗口";
+        } else {
+            title.innerText = "出现错误";
+            title.style.color = "red";
+            content.innerText = status.message;
+            if (status.stacktrace) {
+                content.style.marginBottom = "1em";
+                stacktrace.innerText = status.stacktrace;
+            }
+        }
+        panel.style.animationPlayState = "running";
+    });
+</script>
+</body>
+</html>
```
打包文件应当执行 Minification 操作，即压缩为最小化文件格式，去除无用换行、空格和缩进

Pigeon0v0 replied:
这边得找@ruattd 可能

ruattd replied:
源代码要保证可读性决定了不可能在这里直接扔 minify 出来的文件，以及这么短的文件在加载性能上有无 minify 几乎没有区别，如果一定要加的话，有人愿意去写个 minify 脚本吗，可以把它加到 workflow 里面实现 ci 自动 minify 并替换掉原来的文件

MoYuan-CN commented on Plain Craft Launcher 2/Pages/PageSetup/PageSetupLink.xaml:
```diff
@@ -6,24 +6,98 @@
                         </Grid.RowDefinitions>
-                        <TextBlock Grid.Row="0" VerticalAlignment="Center" HorizontalAlignment="Left" Text="中继服务器地址" Margin="0,0,25,0" />
-                        <local:MyTextBox x:Name="TextLinkRelay" Grid.Row="0" Grid.ColumnSpan="2" Tag="LinkRelay" Grid.Column="1">
+                        <TextBlock Grid.Row="0" VerticalAlignment="Center" HorizontalAlignment="Left" Text="自定义节点" Margin="0,0,25,0" />
```
```suggestion
                        <TextBlock Grid.Row="0" VerticalAlignment="Center" HorizontalAlignment="Left" Text="自定义中继节点" Margin="0,0,25,0" />
```

Pigeon0v0 replied:
刻意的行为，因为有些节点不能中继

Pigeon0v0 committed:
chore: remove unused dll

MoYuan-CN commented on Plain Craft Launcher 2/Pages/PageSetup/PageSetupLink.xaml:
```diff
@@ -6,24 +6,98 @@
+                            <RowDefinition Height="30" />
+                        </Grid.RowDefinitions>
+                        <TextBlock x:Name="TextUsername" Grid.Row="0" VerticalAlignment="Center" HorizontalAlignment="Left" 
+                                   Text="已以 田所浩二 的身份登录至 Natayark Network" />
```
```suggestion
                                   Text="已以 玩家 的身份登录至 Natayark Network" />
```

~~不要在代码里面玩梗~~

MoYuan-CN commented on Plain Craft Launcher 2/Pages/PageLink/PageLinkFeedback.xaml:
```diff
@@ -6,24 +6,50 @@
     <local:MyScrollViewer VerticalScrollBarVisibility="Auto" HorizontalScrollBarVisibility="Disabled" x:Name="PanBack">
         <StackPanel Name="PanMain" Margin="25,25,25,10" Grid.IsSharedSizeScope="True">
-            <local:MyCard Title="关于" Margin="0,0,0,15">
+            <local:MyCard Title="关于与使用条款" Margin="0,0,0,15">
```
```suggestion
            <local:MyCard Title="说明与条款" Margin="0,0,0,15">
```

Pigeon0v0 committed:
ui: apply suggestion

Co-authored-by: 墨渊 <m56042764@163.com>

Pigeon0v0 committed:
ui: apply suggestion

Co-authored-by: 墨渊 <m56042764@163.com>

Pigeon0v0 committed:
feat: 允许对节点进行选择
fix: 刷新按钮自添加以来没好使过

Pigeon0v0 committed:
Merge branch 'link-port' of https://github.com/PCL-Community/PCL2-CE-Pigeon into link-port

Pigeon0v0 committed:
Merge branch 'dev' into link-port
